ludwig/schema/preprocessing.py

```diff
@@ -0,0 +1,751 @@
+        default=None,
+        allow_none=True,
+        description="This can be either the name of a pretrained HuggingFace model or a path where it was downloaded",
+    )
+
+    tokenizer: Optional[str] = schema_utils.String(
+        default="space_punct",
+        allow_none=False,
+        description="Defines how to map from the raw string content of the dataset column to a sequence of elements.",
+    )
+
+    vocab_file: Optional[str] = schema_utils.String(
+        default=None,
+        allow_none=True,
+        description="Filepath string to a UTF-8 encoded file containing the sequence's vocabulary. On each line the "
+        "first string until \t or \n is considered a word.",
+    )
+
+    max_sequence_length: Optional[int] = schema_utils.PositiveInteger(
+        default=256,
+        allow_none=False,
+        description="The maximum length (number of tokens) of the text. Texts that are longer than this value will be "
+        "truncated, while texts that are shorter will be padded.",
+    )
+
+    most_common: Optional[int] = schema_utils.PositiveInteger(
+        default=20000,
+        allow_none=False,
+        description="The maximum number of most common tokens in the vocabulary. If the data contains more than this "
+        "amount, the most infrequent symbols will be treated as unknown.",
+    )
+
+    padding_symbol: Optional[str] = schema_utils.String(
+        default="<PAD>",
+        allow_none=False,
+        description="The string used as a padding symbol. This special token is mapped to the integer ID 0 in the "
+        "vocabulary.",
+    )
+
+    unknown_symbol: Optional[str] = schema_utils.String(
+        default="<UNK>",
+        allow_none=False,
+        description="The string used as an unknown placeholder. This special token is mapped to the integer ID 1 in "
+        "the vocabulary.",
+    )
+
+    padding: Optional[str] = schema_utils.StringOptions(
+        ["left", "right"],
+        default="right",
+        allow_none=False,
+        description="the direction of the padding. right and left are available options.",
+    )
+
+    lowercase: Optional[bool] = schema_utils.Boolean(
+        default=True,
+        description="If true, converts the string to lowercase before tokenizing.",
+    )
+
+    missing_value_strategy: Optional[str] = schema_utils.StringOptions(
+        ["fill_with_const", "fill_with_mode", "fill_with_mean", "backfill", "bfill", "pad", "ffill", "drop_row"],
+        default="fill_with_const",
+        allow_none=False,
+        description="What strategy to follow when there's a missing value in a text column",
+    )
+
+    fill_value: Optional[str] = schema_utils.String(
+        default="<UNK>",
+        allow_none=False,
+        description="The value to replace the missing values with in case the missing_value_strategy is fill_value",
+    )
+
+    computed_fill_value: Optional[str] = schema_utils.String(
```

Comment:
What is the difference between `fill_value` and `computed_fill_value`?

It looks like `computed_fill_value` might be an internal-only parameter that perhaps we want to leave out of the schema.

Reply:
I believe it is just an internal parameter. Specifically, I think it is the fill_value that we calculate based on the data internally. However, throughout the original schema implementation and all of the testing utils, they have `computed_fill_value` as a member of the schema so I figured I would stay consistent and keep it in. I think the best thing to do hear would be to ask #ludwig channel and see why it's in the schema, and if theres not a strong reason to keep in the schema, I can remove it throughout the entire project.

Reply:
Checked with Piero and confirmed it's internal only, so removing from schema.

Reply:
Now moving to `parameter_metadata` schema validation method with `internal_only` flag.